### PR TITLE
Feat/logger and api logger

### PR DESCRIPTION
**Pull request by kwajiehao**
## Overview

This PR goes some way to addressing the logging needs of the project. It accomplishes the following:
1. Add a `CloudWatchLogger` class which uses the `Winston` package to log directly to CloudWatch
2. Add an `apiLogger` middleware which logs all API requests to our server
3. Retrieve a user's GitHub user ID during the authentication phase and attaches it to their JWT

These changes help with log reliability, auditing, and log identifiability.
- **Reliability**: elastic beanstalk instances batch log statements before sending them to CloudWatch with its own daemon process. If the instance dies, then we might lose logs which have been batched up but not sent. Using an external logger such as Winston allows us to log directly to CloudWatch instantaneously instead of being afraid of losing logs when an instance dies.
- **Audit**: by logging every API request made, we can maintain an audit log of what every user has done, and when. This is important since the government often conducts audits and needs audit logs for administrative purposes.
- **Identifiability**: currently, we are unable to identify users from their API requests, since the only information we have about them is their access token, which is a random jumble of characters. By retrieving their user id, we can reconcile actions taken with the identity of the user taking that action, which helps with the audit process.

### CODE REVIEW

**Review by alexanderleegs — CHANGES_REQUESTED** [review]
Left comments. Also, regarding the information that we log in the API logger - does the request body have any important information we should be logging? It's mostly made of encrypted content, which may not be very useful in debugging - let me know what you think about this

**Comment by kwajiehao** [comment]
> Left comments. Also, regarding the information that we log in the API logger - does the request body have any important information we should be logging? It's mostly made of encrypted content, which may not be very useful in debugging - let me know what you think about this

Removed logging of request body in [71f2ffc](https://github.com/isomerpages/isomercms-backend/pull/69/commits/71f2ffcdf30d1911588d3dedb6bdd55a7e2af538)

File: middleware/auth.js [hunk 1]
```diff
@@ -34,7 +38,7 @@ const verifyJwt = (req, res, next) => {
 
 // Login and logout
 auth.get('/auth', noVerify)
-auth.get('/auth/logout', noVerify)
+auth.get('/auth/logout', verifyJwt)
```
[the remark this answers is not in the document]
Comment (kwajiehao) [reply middleware/auth.js]: nice catch. fixed with [02cbdde](https://github.com/isomerpages/isomercms-backend/pull/69/commits/02cbdde0c1a465b5c0dbcc38aa0be193096fc4e5)

**Review by alexanderleegs — APPROVED** [review]
lgtm

**Pull request merged**